File: lab1/fp_SimUDuck/ModelDuck.cpp
#include "stdafx.h"
#include "ModelDuck.h"

ModelDuck::ModelDuck()
	: Duck(FlyBehaviors::GetFlyNoWay(),
		QuackBehaviors::GetQuackBehavior(),
		DanceBehaviors::GetNoDanceBehavior())
{
}

void ModelDuck::Display() const
{
	std::cout << "I'm model duck" << std::endl;
}

File: lw1_fn/src/duck/ModelDuck.php
<?php

class ModelDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            FlyBehavior\getFlyNoWay(),
            QuackBehavior\getQuackBehavior(),
            DanceBehavior\getNoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m model duck!' . PHP_EOL;
    }
}